Sources/DeepLinking/Router.swift

```diff
@@ -9,7 +9,9 @@
 import UIKit
 
 public protocol RouteHandler {
-    func handle(route: URL, parameters: [String : String], queryItems: [URLQueryItem])
+    associatedtype T
+
+    func handle(route: URL, parameters: [String : String], queryItems: [URLQueryItem], completion: ((T) -> Void)?)
```

Comment:
Another important aspect is that the user isn't supposed to call `Handler.handle` directly. That's the `TreeRouter`'s responsibility, and as such having a default parameter on implementations isn't relevant since `TreeRouter.route` simply forwards whatever `completion` it receives to the respective handler. 😉